src/main/java/com/linkedin/kafka/clients/largemessage/LargeMessage.java

```diff
@@ -17,15 +17,16 @@
  */
 public class LargeMessage {
   private final Map<Integer, ByteBuffer> _segments;
-  private final int _messageSize;
+  private final int _originalValueSize;
```

Comment:
As someone that had not read very much of the internals of Kafka I found the term "message" very confusing because (as far a I can tell) there is no such class.  Even the concept of message is fuzzy.  Does it mean just the value?  Or the key and the value?  What about the timestamp, headers and anything else that gets added to the ConsumerRecord or ProducerRecord class in the future?  So does message mean record?  We could serialize the entire record and then chunk that into smaller segments.  What is true is that large message support actually means sending large values that are part of records.